### PR TITLE
use exclude_filename in dist.ini

### PR DESCRIPTION
**Pull request by atoomic**
Avoid using exclude_match which can be too greedy.
All these files exists at the root of the distro.
Add an anchor when exclude_match is required.

Check that MANIFEST file is the same before/after.

built on top of #309 following @Grinnz suggestion

### CODE REVIEW

File: dist.ini [hunk 1]
```diff
@@ -8,18 +8,21 @@ main_module = lib/Paws.pm
...
-exclude_match=botocore
-exclude_match=Makefile
+exclude_match    = ^old_t
+exclude_filename = TODO
+exclude_filename = TODO_in_branch
+exclude_match    = ^templates
+exclude_match    = ^auto-lib
+exclude_match    = ^builder-lib
+exclude_match    = ^builder-bin
+exclude_filename = botocore
```
Comment (Grinnz) [inline dist.ini]: This should probably also be a match as botocore is a submodule

Comment (atoomic) [reply dist.ini]: can update it to be a match but as it stands the MANIFEST is the same

Comment (atoomic) [reply dist.ini]: both works pushing the update with a match for botocore

**Comment by pplu** [comment]
Released to CPAN with Paws 0.39. Thanks for your contribution 😄

**Pull request merged**